### ISSUE TITLE
"Lone" points after triangulation

### ISSUE THREAD

**Issue by mirecc414** [abellgithub/delaunator-cpp#23]
Hello, I am running delaunator on set of points to determine set of adjacent points for each point. Here is code sample:

```
static void triangulate(std::vector<MapNode>* nodes)
{
	//Run Delauney triangulation to figure out adjacent elements for traversing
	std::vector<double> coords;
	for (auto p : *nodes)
	{
		coords.push_back(p.pos.x);
		coords.push_back(p.pos.y);
	}

	delaunator::Delaunator del = delaunator::Delaunator(coords);
	//clear previous adjacency data
	for (MapNode node : *nodes)
	{
		node.adjacent.clear();
	}

	for (int i = 0; i < del.triangles.size(); i++)
	{
		int pointId = del.triangles[i];
		size_t heId = del.halfedges[nextHalfedge(i)];
		if (heId != std::numeric_limits<std::size_t>::max())
		{
			nodes->at(pointId).adjacent.push_back(&nodes->at(del.triangles[heId]));
		}
	}
}
```

After this procedure I end up with some of points without any connections to adjacent points. I read something about delaunator ignoring some close positioned points during triangulation. Could that be explanation for my problem?
In that case it would be possible to discard said points without any connections, to solve my problem.

Thanks

**Comment by abellgithub** [comment]
I have no idea what `nextHalfedge` is. There should be three halfedges for each triangle. Edges that are on the convex hull of the triangulation have no adjacent triangle so have the value INVALID_INDEX.

**Comment by abellgithub** [comment]
I added to the basic example [here.](https://github.com/abellgithub/delaunator-cpp/blob/master/examples/basic.cpp)

**Comment by mirecc414** [comment]
Hello, I have solved my problem so maybe someone will find it useful:
Problem was that some points from input seemed not included in triangulation. Note that input for triangulation is not some predetermined set of points but I am randomly generating those on fly. After some debugging I found out that I had some point overlapping, and those were skipped and did not make it into triangulation. So make sure that your points don't overlap...
